### PR TITLE
Improve Docker Build+Testing

### PR DESCRIPTION
**Pull request by ImreSamu**
Improve Docker Build+Testing

####   Alpine/Ubuntu Dockerfiles:
-  Lot of small updates .
-  New Build Options:
   - DEBUG_MODE (default: false): Option to preserve the build environment for debugging.
     
####  New  `./ci/docker_matrix.sh` for enhanced testing. ( experimental )
* Supports multiple base images (Ubuntu, Alpine).
* Allows testing across various PostgreSQL versions and compilers.
* Includes debug mode and dry-run capability.     

#### Update CI Workflow (`dockertest.yml`) 
* Refactored test configurations in the CI matrix.
* Added a regression flag to control test execution.
* Integrated the official PostgreSQL Docker test suite.

#### Handle Missing COMMIT_HASH in Docker Builds 
 * Implemented a fake COMMIT_HASH (000...000) when .git history is unavailable (e.g., in Docker builds from tarballs).
 * Ensured that regression tests for orioledb_commit_hash() pass without Git history
 
+ Updated docker related docs .

+ see commit messages 

EDIT:
*  Set PostgreSQL 17 - as a default ( Dockerfiles + Docker related docs, scripts  )

### CODE REVIEW

**Comment by za-arthur** [comment]
@ImreSamu thank you for the PR! Could you check the tests? Two tests which use `RUN_REGRESSION_TESTS=yes` don't pass.

**Comment by ImreSamu** [comment]
Ready to review.

File: test/README.md [hunk 1]
```diff
@@ -1,15 +1,13 @@
...
 see: "Docker Official Images Test Suite":
 * https://github.com/docker-library/official-images/tree/master/test
 
-
-used by:
-* `./ci/local_docker_matrix.sh`
-* todo: add to Github CI
-
+Used by:
+* `./docker_matrix.sh`
```
Comment (za-arthur) [inline test/README.md]:
```suggestion
* `./ci/docker_matrix.sh`
```

**Comment by za-arthur** [comment]
Thanks for the fixes. I see that you removed possibility of running regression tests. But I think it can be added later if necessary.

I checked `/.ci/docker_matrix.sh` on a Linux host and on a macOS host (I had to install `bash` and `gnu-getopt` using brew) and I get expected output.

I left just one suggestion above. Could you also make `17` version as a default version? `main` branch of OrioleDB now supports PostgreSQL 17.

File: doc/docker_usage.md [hunk 2]
```diff
@@ -160,61 +203,100 @@ Read more:  https://github.com/docker-library/docs/blob/master/postgres/README.m
...
+  - https://hub.docker.com/_/ubuntu
+- Supported base image architectures:
+  - [ amd64, arm32v7, arm64v8, ppc64le, riscv64, s390x]
+
+##### Alpine
+- Supported base images (with security updates):
+  - https://hub.docker.com/_/alpine
+- Supported base image architectures:
+  - [ amd64, arm32v6, arm32v7, arm64v8, i386, ppc64le, riscv64, s390x ]
+
```
Comment (za-arthur) [inline doc/docker_usage.md:281]:
````suggestion

#### macOS
On macOS you might need to install `bash` and `gnu-getopt` from Homebrew. To install them run the command:

```bash
brew install bash gnu-getopt
```

Update your `/etc/shells`:

```bash
echo /opt/homebrew/bin/bash >> /etc/shells
```

You may need to update your `PATH` variable in the `.bashrc` or `.zshrc` file:

```
PATH=/opt/homebrew/bin:/opt/homebrew/opt/gnu-getopt/bin:$PATH
```

````

**Comment by ImreSamu** [comment]
Ready to review ,

>  I see that you removed possibility of running regression tests. But I think it can be added later if necessary.

Yes, I haven’t abandoned the idea entirely. 
Currently, however, the `"toast"` test isn’t running successfully on my end, 
as `"test_decoding"` doesn’t seem to install for some reason, and I haven’t yet figured out why.

I'm getting the error:
```
--- /usr/src/postgresql/contrib/orioledb/expected/toast.out	2024-10-09 12:33:59.233743300 +0000
+++ /usr/src/postgresql/contrib/orioledb/results/toast.out	2024-10-26 20:28:44.579171105 +0000
@@ -1244,11 +1244,7 @@
 -- Uncompressed
 CREATE TABLE o_logical(id integer PRIMARY KEY, v1 text, v2 text) using orioledb WITH (compress = -1, toast_compress = -1, primary_compress = -1);
 SELECT slot_name FROM pg_create_logical_replication_slot('regression_slot', 'test_decoding', false, true);
-    slot_name    
------------------
- regression_slot
-(1 row)
-
+ERROR:  could not access file "test_decoding": No such file or directory
```

I plan to run tests on the Orioledb Docker images produced by the docker build process, rather than during the actual Docker build itself.
- This will simplify the Dockerfile.
- Additionally, I can run the tests in `--privileged` mode, which isn’t possible during the build stage.

In any case, once I have this set up, I’ll create a follow-up PR.

**Comment by akorotkov** [comment]
Merged. Thank to @ImreSamu for the work, and to @za-arthur for the review!

**Pull request merged**